OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -46,6 +46,21 @@ public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttribute
             UserId = userId;
         }
 
+        public Optimizely GetOptimizely()
+        {
+            return Optimizely;
+        }
+
+        public string GetUserId()
+        {
+            return UserId;
+        }
+
+        public UserAttributes GetAttributes()
+        {
+            return new UserAttributes(Attributes);
```

Comment:
Don't we need a mutex protection for this copy with the same lock for setAttribute?